File: Medium/151_Reverse_Words_in_a_String/Solution.java



public class Solution {
    public String reverseWords(String s) {
        String[] words = s.trim().split("\\s+");
        StringBuilder sb = new StringBuilder();
        
        for (int i = words.length - 1; i >= 0; i--) {
            sb.append(words[i]);
            if (i > 0) {
                sb.append(" ");
            }
        }
        
        return sb.toString();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        String s1 = "the sky is blue";
        System.out.println(solution.reverseWords(s1)); // Output: "blue is sky the"
        
        // test cases 2
        String s2 = "  hello world  ";
        System.out.println(solution.reverseWords(s2)); // Output: "world hello"
    }
}

File: Medium/151_Reverse_Words_in_a_String/Solution.py
from typing import List, Optional


class Solution(object):
    def reverseWords(self, s: str) -> str:
        l = (s.split(" "))

        while ("" in l):
            l.remove("")
        
        l = l[::-1]

        ans = ""
        for i in l:
            ans += (i + " ")

        return ans[:-1]


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    s = "the sky is blue"
    print(solution.reverseWords(s))  # Output: "blue is sky the"

    # test cases 2
    s = "  hello world  "
    print(solution.reverseWords(s))  # Output: "world hello"